core/server_config.go

```diff
@@ -90,8 +126,22 @@ type HTTPConfig struct {
 	Address string `koanf:"address"`
 	// CORS holds the configuration for Cross Origin Resource Sharing.
 	CORS HTTPCORSConfig `koanf:"cors"`
+	// TLSMode specifies whether TLS is enabled for this interface, and which flavor.
+	TLSMode HTTPTLSMode `koanf:"tls"`
 }
 
+// HTTPTLSMode defines the values for TLS modes
+type HTTPTLSMode string
+
+const (
+	// DisabledHTTPTLSMode specifies that TLS is not enabled for this interface.
+	DisabledHTTPTLSMode HTTPTLSMode = "disabled"
+	// ServerCertTLSMode specifies that TLS is enabled for this interface, but no client certificate is required.
+	ServerCertTLSMode = "server-cert"
+	// MutualTLSMode specifies that TLS is enabled for this interface, and that it will require a client certificate.
+	MutualTLSMode = "server-and-client-cert"
```

Comment:
```suggestion
	// DisabledHTTPTLSMode specifies that TLS is not enabled for this interface.
	DisabledHTTPTLSMode HTTPTLSMode = "disabled"
	// ServerCertTLSMode specifies that TLS is enabled for this interface, but no client certificate is required.
	ServerCertTLSMode HTTPTLSMode = "server-cert"
	// MutualTLSMode specifies that TLS is enabled for this interface, and that it will require a client certificate.
	MutualTLSMode HTTPTLSMode = "server-and-client-cert"
```